specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+# Knative Control Plane Contract
+## Abstract
+
+The Knative Eventing platform provides common primitives for routing CloudEvents
+between cooperating HTTP clients. This document describes the structure,
+lifecycle, and management of Knative Eventing resources in the context of the
+[Kubernetes Resource Model](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/resource-management.md).
+An understanding of the Kubernetes API interface and the capabilities of
+[Kubernetes Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+is assumed.
+
+This document does not define the
+[data plane event delivery contract](./data-plane.md) (though it does describe
+how event delivery is configured). This document also does not prescribe
+specific implementations of supporting services such as access control,
+observability, or resource management.
+
+## Background
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+There is no formal specification of the Kubernetes API and Resource Model. This
+document assumes Kubernetes 1.21 behavior; this behavior will typically be
```

Comment:
I wonder if we need to say something about the "kube 1.21" statement? For example, when/how often, will this version number change?

Reply:
I don't know. Kube API changes have gotten _less_ disruptive over time, but the most recent "interesting" change was `managedFields` in 1.16.

Reply:
yes "interesting" is a good word for it :-)